src/growl.c

```diff
@@ -301,13 +303,17 @@ growl_tcp_register(
             goto leave;
         } else {
             int len = strlen(line);
-            /*fprintf(stderr, "%s\n", line);*/
-            if (strncmp(line, "GNTP/1.0 -ERROR", 15) == 0) {
-                if (strncmp(line + 15, " NONE", 5) != 0) {
-                    fprintf(stderr, "failed to register notification\n");
+            if (!strncmp(line, ERRSTR, strlen(ERRSTR))) {
+                fprintf(stderr, "Failed to register notification:\n%s\n", line);
+                free(line);
+                //Print full error-message
+                while((line = growl_tcp_read(sock)) != 0){
+                    fprintf(stderr, "%s\n", line);
+                    int br = (strlen(line) == 0);
```

Comment:
@TheOneRing I fell for this one in one of my first versions, too. growl_tcp_read returns NULL on an error, but a string of length 0 if there is no more data. I admit, though, that the code is not very nice. I'll fix it to 
```
char firstchar = line[0];
free(line);
if(firstchar == '\0') break;
```
as soon as I have access to my development system.